pkg/cloud/flush_writer.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package cloud
+
+import (
+	"context"
+	"io"
+
+	"github.com/cockroachdb/pebble/objstorage"
+)
+
+// PebbleWriter is a wrapper around a cloud writer that implements the pebble
+// objstorage.Writable API.
+type PebbleWriter struct {
```

Comment:
nit: I'd lean towards AbortableWriter since there's nothing pebble-specific about the actual writing. I'd just as soon keep `pkg/cloud` a more generic utility for moving bytes to/from cloud services and if it just so happens its API matches that which pebble needs, great?

Reply:
Done.